File: docs/guides/swift/Sources/searchWithRuleContextBuyer.swift
import Foundation
#if os(Linux) // For linux interop
    import FoundationNetworking
#endif

import Core
import Search

let getBuyerAccountId = { "" } // Implement your logic here

func searchWithRuleContextBuyer() async throws {
    let client = try SearchClient(appID: "ALGOLIA_APPLICATION_ID", apiKey: "ALGOLIA_API_KEY")

    // get the buyer account information
    let buyer = getBuyerAccountId()
    let searchParams = SearchSearchParams.searchSearchParamsObject(
        SearchSearchParamsObject(query: "<YOUR_SEARCH_QUERY>", ruleContexts: [buyer])
    )

    let response: SearchResponse<Hit> = try await client.searchSingleIndex(
        indexName: "<YOUR_INDEX_NAME>",
        searchParams: searchParams
    )
    print(response)
}

File: docs/guides/kotlin/src/main/kotlin/com/algolia/snippets/searchWithRuleContextBuyer.kt
import com.algolia.client.api.SearchClient
import com.algolia.client.configuration.*
import com.algolia.client.extensions.*
import com.algolia.client.model.search.*
import com.algolia.client.transport.*

val getBuyerAccountId: () -> String = {
  "" // Implement your logic here
}

suspend fun searchWithRuleContextBuyer() {
  val client = SearchClient(appId = "ALGOLIA_APPLICATION_ID", apiKey = "ALGOLIA_API_KEY")

  // get the buyer account information
  val buyer = getBuyerAccountId()
  val searchParams = SearchParamsObject(query = "<YOUR_SEARCH_QUERY>", ruleContexts = listOf(buyer))

  client.searchSingleIndex(indexName = "<YOUR_INDEX_NAME>", searchParams = searchParams)
}
